src/client/components/pages/parts/collections-table.js

```diff
@@ -0,0 +1,173 @@
+ */
+
+import * as bootstrap from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import React from 'react';
+import _ from 'lodash';
+import {genEntityIconHTMLElement} from '../../../helpers/entity';
+
+
+const {DropdownButton, MenuItem, Table} = bootstrap;
+
+
+class CollectionsTable extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			type: ''
+		};
+
+		// React does not autobind non-React class methods
+		this.handleEntitySelect = this.handleEntitySelect.bind(this);
+	}
+
+	handleEntitySelect(type) {
+		this.setState({type});
+		this.props.onTypeChange(type);
+	}
+
+	render() {
+		const {showOwner, showOwnerCollaborator, showPrivacy, results, tableHeading} = this.props;
+		const entityTypeSelect = (
+			<DropdownButton
+				bsStyle="primary"
+				id="entity-type-select"
+				title={_.startCase(this.state.type) || 'Entity Type'}
+				onSelect={this.handleEntitySelect}
+			>
+				{this.props.entityTypes.map((entityType) => (
+					<MenuItem
+						eventKey={entityType}
+						key={entityType}
+					>
+						{genEntityIconHTMLElement(entityType)}
+						{_.startCase(entityType)}
+					</MenuItem>
+				))}
+				<MenuItem divider/>
+				<MenuItem
+					eventKey="all_types"
+					key="allTypes"
+				>
+					All Types
+				</MenuItem>
+			</DropdownButton>
+		);
+		return (
+			<div>
+				<div>
+					<h1 className="text-center">
+						{tableHeading}
+					</h1>
+					<div className="text-right">
+						{entityTypeSelect}
+					</div>
+				</div>
+				<hr className="thin"/>
+				{
+					results.length > 0 ?
+						<Table
+							responsive
+							className="table table-striped"
+						>
+							<thead>
+								<tr>
+									<th className="col-sm-2">Name</th>
+									<th className="col-sm-4">Description</th>
+									<th className="col-sm-2">Entity Type</th>
+									{
+										showPrivacy ?
+											<th className="col-sm-2">Privacy</th> : null
+									}
+									{
+										showOwnerCollaborator ?
+											<th className="col-sm-2">Collaborator/Owner</th> : null
+									}
+									{
+										showOwner ?
+											<th className="col-sm-2">Owner</th> : null
+
+									}
+								</tr>
+							</thead>
+
+							<tbody>
+								{
+									results.map((collection) => (
+										<tr key={collection.id}>
+											<td>
+												<a
+													href={`/collection/${collection.id}`}
+												>
+													{collection.name}
+												</a>
+											</td>
+											<td>{collection.description}</td>
+											<td>{collection.entityType}</td>
+											{
+												showPrivacy ?
+													<td>{collection.public ? 'Public' : 'Private'}</td> : null
+											}
+											{
+												showOwnerCollaborator ?
+													<td>{collection.isOwner ? 'Owner' : 'Collaborator'}</td> : null
+											}
+											{
+												showOwner ?
+													<td>{collection.owner.name}</td> : null
+
+											}
+										</tr>
```

Comment:
What do you think about showing the last modified datetime here, using the formatDate utility?